diff --git a/src/hb-algs.hh b/src/hb-algs.hh
index 007db9e4..ec0ff610 100644
--- a/src/hb-algs.hh
+++ b/src/hb-algs.hh
@@ -167,10 +167,22 @@ template <unsigned Pos=1, typename Appl, typename V>
 auto hb_partial (Appl&& a, V&& v) HB_AUTO_RETURN
 (( hb_partial_t<Pos, Appl, V> (a, v) ))
 
+/* The following hacky replacement version is to make Visual Stuiod build:. */ \
+/* https://github.com/harfbuzz/harfbuzz/issues/1730 */ \
+#ifdef _MSC_VER
+#define HB_PARTIALIZE(Pos)                              \
+  template <typename _T>                                \
+  decltype (auto) operator() (_T &&_v) const            \
+  {                                                     \
+    return hb_partial<Pos> (this, hb_forward<_T> (_v)); \
+  }                                                     \
+  static_assert (true, "")
+#else
 #define HB_PARTIALIZE(Pos) \
   template <typename _T> \
   auto operator () (_T&& _v) const HB_AUTO_RETURN (hb_partial<Pos> (this, hb_forward<_T> (_v))) \
   static_assert (true, "")
+#endif
 
 
 struct
